### DISCUSSION TITLE
Large force RMSE

### DISCUSSION THREAD

**Discussion by melicherova** [deepmodeling/deepmd-kit#1577] [Q&A]
Hi,
I have DFT MD trajectories from melting crystal nitrogen and liquid cooling under pressure
and I want to train NN with the data from this simulation.

But I have problem with large force RMSE in trainning which is 913 meV/A (energy RMSE is quite reasonable 9 meV/atom).
I used trained NN in lammps MD and I tried to reproduce protocol from DFT MD but crystal melted at much lower temperature
then in DFT MD.

Are there some options how to reduce force RMSE?

Thanks

I am using deepmd-kit  v2.0.3 and input for deepmd is:

    "model": {
        "type_map":     ["N"],
        "descriptor" :{
            "type":             "se_e2_a",
            "sel":              [190],
            "rcut_smth":   5.80,
            "rcut":             6.00,
            "neuron":           [25, 50, 100],
            "resnet_dt":        false,
            "axis_neuron":      16,
            "seed":             1,
            "_comment":         " that's all"
        },
        "fitting_net" : {
            "neuron":           [240, 240, 240],
            "resnet_dt":        true,
            "seed":             1,
            "_comment":         " that's all"
        },
        "_comment":     " that's all"
    },

    "learning_rate" :{
        "type":         "exp",
        "decay_steps":  5000,
        "start_lr":     0.001,
        "stop_lr":      3.51e-8,
        "_comment":     "that's all"
    },

    "loss" :{
        "type":         "ener",
        "start_pref_e": 0.02,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 0.2,
        "limit_pref_v": 1,
        "_comment":     " that's all"
    },

**Comment by McAlexDudu** [comment]
I also have this kind of question. My force RMSE is 55 meV/A which should at most 5 meV/A. I use dpgen can solve the problem; but I also want to know how to decrease the RMSE while using just deepmd. I have tried larger sel but not work.

By the way, my rmse_f_val from lcurve.out seem to almost keep in 40~60 meV/A from after 10000 steps, and seem to not converged?

**Reply by RobinMAJ** [reply]
Dear McAlexDudu 
   Could you provide the way  on how dpgen solves this problem? Thank you very much!!

**Reply by McAlexDudu** [reply]
@RobinMAJ I think my problem maybe the dataset not set right, or my AIMD data were not good. During dpgen, I follow the step of init_bulk to create dataset, and the RMSE can decrease to at least 4 meV/A of four models.

**Comment by njzjz** [comment]
Could you provide the scatter plot (DFT vs NN)? I suggest that training data is extremely wrong.

**Reply by melicherova** [reply]
Yes, I attach a correlation plot.
In what sense can be training data wrong?
The training data used is converged.
I've already wondered what the problem with the data might be. Perhaps, various training data is the problem (I have a lot of liquid structures in training data set) then the forces are quite large.


![nitrogen](https://user-images.githubusercontent.com/78050977/158956306-512a78e1-e2b3-4526-862f-a61ec56eb695.png)

**Reply by njzjz** [reply]
I don't find anything wrong in the figure. Did you train enough steps?

**Reply by melicherova** [reply]
I used 1 000 000 training steps.

**Reply by njzjz** [reply]
Does your DFT data have enough accuracy? For example, all of the systems are converged to within 1e-7 Hartree.

**Reply by wanghan-iapcm** [reply]
The standard deviation of DFT forces look very large (~20 eV/A ?). When we talk about the accuracy of the model, we compare the error with respect to the standard deviation in the label. In your case the force error is about 1/10 of the standard deviation, your model has accuracy.

**Reply by melicherova** [reply]
I plotted a histogram of  DFT forces and the standard deviation is not so large. Yes, There are a few forces which are about 20 eV/A but the majority of forces is much smaller and standard deviation is ~ 3 eV/A.
![forces_hist](https://user-images.githubusercontent.com/78050977/159285117-e1872895-4952-4b20-9cee-bdd7e943848f.png)

**Reply by melicherova** [reply]
I have one more observation for my NN model. The RMSE of   crystalline structures is about 340 meV/A and liquid structures are source of bigger force RMSE. Can the problem be insufficient coverage of configuration space in case of liquid structures?

**Reply by njzjz** [reply]
Just a confirmation, did you directly use the coordinates, energies and forces from the MD simulations, but didn't reanalyze frames?

**Reply by melicherova** [reply]
Yes, I used the data directly from MD simulations. What do you imagine under the reanalysis of frames?

**Reply by njzjz** [reply]
If the MD software - I don't know what you used - applies velocity-Verlet integration, the dumped coordinates may be half step ahead of the energies and forces. You need to check it.

**Reply by melicherova** [reply]
I used VASP.

**Reply by njzjz** [reply]
I am not familiar with VASP. You can select a frame and calculate the single-point energy to compare  and see if there's any difference.

**Reply by melicherova** [reply]
I have done it and it should be OK so coordinates, energies and forces are consistent.

**Reply by njzjz** [reply]
That's good. But I think 340 meV/A is still a bit large.
Did you train crystalline structures and liquid structures at the same time?

**Reply by wanghan-iapcm** [reply]
Could you please confirm that the VASP calculation of force is converged with large enough ENCUT and small enough KSPACING?

**Reply by melicherova** [reply]
I have used only gamma point in my calculations because I have quite large supercell with 512 atoms. Results shouldn't change significantly with smaller KSCAPING and I used ENCUT = 1.3xENMAX, it also should be ok. How much it is needed to have forces and energies converged?
I have a lot of liquid polymeric structures in the training data set and this structures are problematic.

**Reply by wanghan-iapcm** [reply]
The value of ENCUT and KSPACING needed to converge the labels depend on systems. We cannot suggest a prior choice (we CANNOT tell if gamma point is enough for a 512 atom system nor encut = 1.3xenmax is large enough). The only thing one can do is the increase the parameters and check the difference in labels. One may not need to do the checks for all the data, but one or two representative frames should be enough.

**Reply by melicherova** [reply]
Thank you, I checked it for two structures and energy deviation is ~ 1e-3 eV/atom, in case of forces rmsd is ~ 1e-2 eV/A with increasing ENCUT . And in case of decreasing kspacing, energy deviation is ~ 1e-4 eV/atom and force deviation is ~ 1e-2 eV/A.

**Reply by wanghan-iapcm** [reply]
That's interesting. The large force error should not come from VASP numerical error. 
I notice you also switched on the virial term in loss. Could you please switch it off and see merely fitting energy and force could reduce the force error?

**Reply by melicherova** [reply]
I switched off virial in loss and force RMSE is now slightly smaller, it is 780 eV/A but still too large. I was thinking if there could be a problem because of triclinic simulation cell which is present in case of liquid and amorphous structures in training data set.

**Reply by wanghan-iapcm** [reply]
Yes that could be a issue. One may want to check the shape of the simulation cell. 

Could you also please provide the learning curves? Please show them in a log-log plot. Thank you!

**Reply by melicherova** [reply]
Yes, there is the learning curves plot.
![lcurve](https://user-images.githubusercontent.com/78050977/160283287-5d6d4be0-9360-4f5d-aba4-2a6174eec212.png)

**Reply by wanghan-iapcm** [reply]
The decaying speed of energy and virial looks perfect. 

Could you also please check the RMAX parameter in your POTCAR and the minimal distance between atoms in the training data? Since your training data is at high pressure, I suspect there is an issue of core overlapping. http://jwz.chem.dal.ca/lab-notes.php

**Reply by melicherova** [reply]
RMAX is 1.529 Bohr and minimal distance between atoms is 1.0008 A in the training data.
However, this short distance comes from very small amount of N2 molecules in liquid structures.
I made two testing calculations with hard pseudopotential with RMAX=1.127. First one was with amorphous nitrogen and there was force deviation   334 meV/A compared to original calculation. Second one was with crystal and force deviation was  272 meV/A. Yes, it is quite large deviation but it does not explain difference in force RMSE in training where force RMSE for crystalline structures is 340 meV/A and for liquid and amorphous structures is ~900 meV/A.

**Reply by wanghan-iapcm** [reply]
The force deviation about ~300 meV/A is significant. The DP model assumes that the force is a derivative of the energy. The error introduced by core-overlapping, however,  does not preserve the correlation.  By far I cannot tell if all the deviation ~300 meV should be attributed to the error. In the worse case, the 300meV is a random error in the force label, it may draw DP from the correct force by 300 meV, and adding the 300meV random error at the test point, this could lead to at most ~600meV error in test. One meaningful test is to exclude all data frames that have short atomic distances, and train the model again to see the error in force. 

From the point of using PP in the vasp calculation, since you have N2 molecules in the data, it is strongly recommended to use the harder PP with RMAC=1.127 Bohr. Of course the convergence with respect to ENCUT should be carefully tested. Harder PP needs larger ENCUT.

**Comment by RobinMAJ** [comment]
Thank you very much!

在2022-03-20 12:46:26，Han ***@***.***写道：

The standard deviation of DFT forces look very large (~20 eV/A ?). When we talk about the accuracy of the model, we compare the error with respect to the standard deviation in the label. In your case the force error is about 1/10 of the standard deviation, your model has accuracy.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
